src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -174,40 +174,26 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, lon
             // Allocate an active request
             QuicStream? quicStream = null;
             Http3RequestStream? requestStream = null;
-            ValueTask waitTask = default;
 
             try
             {
                 try
                 {
-                    while (true)
+                    if (_connection != null)
                     {
-                        lock (SyncObj)
+                        if (HttpTelemetry.Log.IsEnabled() && queueStartingTimestamp == 0 && _connection.GetRemoteAvailableBidirectionalStreamCount() == 0)
                         {
-                            if (_connection == null)
-                            {
-                                break;
-                            }
-
-                            if (_connection.GetRemoteAvailableBidirectionalStreamCount() > 0)
-                            {
-                                quicStream = _connection.OpenBidirectionalStream();
-                                requestStream = new Http3RequestStream(request, this, quicStream);
-                                _activeRequests.Add(quicStream, requestStream);
-                                break;
-                            }
-
-                            waitTask = _connection.WaitForAvailableBidirectionalStreamsAsync(cancellationToken);
+                            // the call below will almost certainly block, measure waiting time for telemetry purposes
+                            queueStartingTimestamp = Stopwatch.GetTimestamp();
                         }
 
-                        if (HttpTelemetry.Log.IsEnabled() && !waitTask.IsCompleted && queueStartingTimestamp == 0)
+                        quicStream = await _connection.OpenBidirectionalStreamAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Can `_connection` become null at this point? I assume it can (we removed the lock around). It might be sufficient to just add `?.` so `quicStream` will remain null in that case.

Reply:
This gets me
```
error CS1061: (NETCORE_ENGINEERING_TELEMETRY=Build) 'ConfiguredValueTaskAwaitable<QuicStream>?' does not contain a definition for 'GetAwaiter'
```

Reply:
That might get us a closed or disposed connection, while before, in case of goaway, it would either be a good open connection, or an already nullified reference.

This would mean weird exceptions would bubble up that would not be wrapped in HttpRequestException+RetryOnConnectionFailure, so the request will not be retried automatically. That sounds bad. cc @ManickaP 

We need to think how to overcome this.

Reply:
Similar problem was lurking there before as well, since we have kicked off an async operation in a lock but we could have disposed/closed the connection before the task was finished. I guess now it's more pronounced.

For now, we can just try/catch around it and in case of an error, leave `quicStream = null` which should eventually translate to retry on different connection few lines bellow.

Reply:
catch block added